server/guides/other/SETUP.md

```diff
@@ -0,0 +1,232 @@
+
+- You can use an in-built linux utility called [tmux](https://github.com/tmux/tmux/wiki). Use this if the server is intended to be temporarily up.
+- You can register minecraft as a **system service**. Use this if the server is intended to be up 24/7 and requires automatic restarting in case of host shutdowns/failures.
+
+#### Setup With Tmux
+
+Start a tmux window
+
+`tmux new-session -t sessionName`
+
+`sessionName` being whatever you want to name the instance, for example `minecraft-session`
+
+This will automatically take you inside the tmux session. Start the server using
+
+`java -Xmx1024M -Xms1024M -jar server-jar nogui`
+
+Exit out of the tmux session using `Ctrl+B` **then** `D`
+
+You server will keep running in the background.
+
+To reattach to the session,
+
+`tmux a -t sessionName`
+
+`sessionName` being whatever you named the instance earlier.
+
+To destroy the session completely, you can attach to the session and then `Ctrl+D` out of it. It detaches from session and destroys it too.
+
+#### Setup As A System Service
+
+You need to set up a service if it should, for example automatically launch at boot or restart when it terminates
+So, create a file named `minecraft.service` in `/etc/systemd/system` (May require root permissions)
+
+--> `sudo touch /etc/systemd/system/minecraft.service` (actually optional)
+
+And write into it:
+
+```txt
+[Unit]
+Description=A Minecraft Server
+After=network.target
+StartLimitIntervalSec=0[Service]
+Type=simple
+Restart=always
+RestartSec=1
+User={YOURUSERNAME}
+ExecStart=/home/{YOURUSERNAME}/server/start.sh
+```
+
+of course, replace `{YOURUSERNAME}` with your username
+
+--> `sudo echo "[Unit]\nDescription=A Minecraft Server\nAfter=network.target\nStartLimitIntervalSec=0[Service]\nType=simple\nRestart=alwaysRestartSec=1\nUser=$USER\nExecStart=$HOME/server/start.sh" > /etc/systemd/system/minecraft.service`
+so you can now start the service
+
+--> `systemctl start minecraft`
+
+Now you should be able to join
+It will restart automatically 1 second after it shut down. To make it start at boot, enable it
+
+--> `systemctl enable minecraft`
+
+# Setup For Docker
+
+#### Prebuilt Docker
+
+> Based on the docker image with the most pulls
+> See [itzg/docker-minecraft-server for more details](https://github.com/itzg/docker-minecraft-server)
+
+##### Simple Launch
+
+```sh
+docker run -d -p 25565:25565 --name mc -e EULA=TRUE itzg/minecraft-server
+```
+
+##### Docker Compose
+
+```yml
+version: "3"
+
+services:
+  mc:
+    image: itzg/minecraft-server
+    ports:
+      - 25565:25565
+    environment:
+      EULA: "TRUE"
+    volumes:
+      # attach the relative directory 'data' to the container's /data path
+      ./data:/data
+```
+
+#### You Build Docker
+
+You can use an already built docker image or just create a Dockerfile with following content:
+
+```dockerfile
+FROM openjdk
+WORKDIR /data
+
+COPY ["server.jar","eula.txt", "start.sh"] ./
+
+EXPOSE 25565
+
+ENTRYPOINT ["/data/start.sh"]
+```
+
```

Comment:
Maybe a reminder to use that in the directory with a server installed and maybe add an option to also copy the `world` folder etc.
For example just copy the whole folder